`build/run make -j2 build.all` failed to build rootfs on arm64

I am wondering if the `build.all` can build all the binaries or container images for arm64 on amd64 host machine. Below is the output of command `build/run make -j2 build.all` on amd64:
```
=== docker build build-c76ef25e/rootfs-builder-amd64
Sending build context to Docker daemon 3.072 kB
Step 1/3 : FROM arm64v8/ubuntu:xenial
xenial: Pulling from arm64v8/ubuntu
39d5c0603809: Pull complete 
4022bcc1888c: Pull complete 
6737bf717a11: Pull complete 
b77eded6bd70: Pull complete 
872459994a86: Pull complete 
Digest: sha256:afcbe0d8dcce89777ea2cfb8eaebe972329c72acdf47ccf85a62ab742d548485
Status: Downloaded newer image for arm64v8/ubuntu:xenial
 ---> 6a0ced0a43d2
Step 2/3 : SHELL ["/bin/bash","-c"]
 ---> Running in 9fe665b66efa
Removing intermediate container 9fe665b66efa
 ---> 01e450f22514
Step 3/3 : RUN sed -i -e 's/^deb-src/#deb-src/' /etc/apt/sources.list &&     DEBIAN_FRONTEND=noninteractive apt-get update &&     DEBIAN_FRONTEND=noninteractive apt-get install -yy --no-install-recommends         net-tools         netcat &&     DEBIAN_FRONTEND=noninteractive apt-get upgrade -yy --no-install-recommends &&     DEBIAN_FRONTEND=noninteractive apt-get autoremove -y &&     DEBIAN_FRONTEND=noninteractive apt-get clean -y &&     rm -rf         /etc/{selinux,systemd,udev}         /lib/{lsb,udev}         /tmp/*         /usr/lib/{locale,systemd,udev}         /usr/share/{doc,info,man}         /var/cache/debconf/*         /var/lib/apt/lists/*         /var/log/*         /var/tmp/*
 ---> Running in 881ab29cc2c0
standard_init_linux.go:195: exec user process caused "exec format error"
The command '/bin/bash -c sed -i -e 's/^deb-src/#deb-src/' /etc/apt/sources.list &&     DEBIAN_FRONTEND=noninteractive apt-get update &&     DEBIAN_FRONTEND=noninteractive apt-get install -yy --no-install-recommends         net-tools         netcat &&     DEBIAN_FRONTEND=noninteractive apt-get upgrade -yy --no-install-recommends &&     DEBIAN_FRONTEND=noninteractive apt-get autoremove -y &&     DEBIAN_FRONTEND=noninteractive apt-get clean -y &&     rm -rf         /etc/{selinux,systemd,udev}         /lib/{lsb,udev}         /tmp/*         /usr/lib/{locale,systemd,udev}         /usr/share/{doc,info,man}         /var/cache/debconf/*         /var/lib/apt/lists/*         /var/log/*         /var/tmp/*' returned a non-zero code: 1
Makefile:37: recipe for target 'rootfs-build' failed
make[2]: *** [rootfs-build] Error 1
Makefile:27: recipe for target 'base.linux_arm64' failed
make[1]: *** [base.linux_arm64] Error 2
make[1]: *** Waiting for unfinished jobs....
sha256:fc6fdb55d46f914bded267f58bdab4f54ee419673a865e61f6e44c95d4e21891
=== docker build build-c76ef25e/rootfs-amd64-fc6fdb55d46f
2099615aba47e8c6e42db10ca01952534792a6fa9c35e55951bab8536f86cf75
sha256:31be09fc0680de8c2627d592fa845be68f7a2c32c23a01a49f57b4b64a381f5e
=== docker build build-c76ef25e/base-amd64
sha256:89ead5c6e4731120f72b0bca84eec05d767c90882b50da245174998d009f53e1
=== caching image build-c76ef25e/base-amd64
=== caching image build-c76ef25e/rootfs-builder-amd64
=== caching image build-c76ef25e/rootfs-amd64-fc6fdb55d46f
Makefile:118: recipe for target 'build.all' failed
make: *** [build.all] Error 2
```
But `build/run make -j2 build` will succeed.

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://Rook-io.slack.com).
3. Did you already search the existing open issues for anything similar? -->

Is this a bug report or feature request?
<!-- Remove only one -->
* Bug Report


**Bug Report**

What happened:
`build/run make -j2 build.all` failed to build `rootfs-build`
What you expected to happen:
`build/run make -j2 build.all` succeed to build `rootfs-build` of arm64
How to reproduce it (minimal and precise):
<!-- Please let us know any circumstances for reproduction of your bug. -->

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
* Kernel (e.g. `uname -a`):
```
Linux dennis-ws 4.15.2 #1 SMP Thu Feb 8 17:40:32 CST 2018 x86_64 x86_64 x86_64 GNU/Linux
```

* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Ceph status (use `ceph health` in the [Rook toolbox](https://Rook.io/docs/Rook/master/toolbox.html)):

--- a/Makefile
+++ b/Makefile
@@ -110,9 +110,9 @@ build: build.common
 	@$(MAKE) go.build
 # if building on non-linux platforms, also build the linux container
 ifneq ($(GOOS),linux)
-	@$(MAKE) go.build PLATFORM=linux_amd64
+	@$(MAKE) go.build PLATFORM=linux_$(GOHOSTARCH)
 endif
-	@$(MAKE) -C images PLATFORM=linux_amd64
+	@$(MAKE) -C images PLATFORM=linux_$(GOHOSTARCH)
 
 build.all: build.common
 	@$(MAKE) do.build.parallel

--- a/build/makelib/golang.mk
+++ b/build/makelib/golang.mk
@@ -172,9 +172,15 @@ go.vendor: $(DEP)
 
 $(DEP):
 	@echo === installing dep
-	@mkdir -p $(TOOLS_HOST_DIR)
-	@curl -sL -o $(DEP) https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-$(GOHOSTOS)-$(GOHOSTARCH)
+	@mkdir -p $(TOOLS_HOST_DIR)/tmp
+	@if [ "$(GOHOSTARCH)" = "arm64" ]; then\
+		GOPATH=$(TOOLS_HOST_DIR)/tmp GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get -u github.com/golang/dep/cmd/dep;\
+		mv $(TOOLS_HOST_DIR)/dep $@;\
+	else \
+		curl -sL -o $(DEP) https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-$(GOHOSTOS)-$(GOHOSTARCH);\
+	fi
 	@chmod +x $(DEP)
+	@rm -fr $(TOOLS_HOST_DIR)/tmp
 
 $(GOLINT):
 	@echo === installing golint

--- a/images/image.mk
+++ b/images/image.mk
@@ -21,10 +21,6 @@ override GOOS=linux
 SELF_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
 include $(SELF_DIR)/../build/makelib/common.mk
 
-ifneq ($(GOHOSTARCH),amd64)
-$(error image build only supported on amd64 host currently)
-endif
-
 # the registry used for cached images
 CACHE_REGISTRY := cache
 

